src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -322,6 +316,55 @@ struct ScalarOp<ONNXTanOp> {
   using IOp = NotSuportedScalarOp;
 };
 
+//===----------------------------------------------------------------------===//
+// Scalar unary ops for lowering ONNXIsInfOp
+//===----------------------------------------------------------------------===//
+template <>
+struct ScalarOp<ONNXIsInfOp> {
+  using FOp = CustomScalarOp;
+  using IOp = NotSuportedScalarOp;
+};
+
+template <>
+Value emitScalarOpFor<ONNXIsInfOp>(ConversionPatternRewriter &rewriter,
+    Location loc, Operation *op, Type elementType,
+    ArrayRef<Value> scalarOperands) {
+
+  Value operand = scalarOperands[0];
+  Type inputElemType = getElementType(operand.getType());
+  Value result;
+
+  CheckIfCustomScalarOpIsSupported<ONNXIsInfOp>(inputElemType);
+  MultiDialectBuilder<MathBuilder> create(rewriter, loc);
+  Value negInf = create.math.negativeInf(inputElemType);
+  Value posInf = create.math.positiveInf(inputElemType);
+
+  double detectNegAttribute = dyn_cast<ONNXIsInfOp>(op).getDetectNegative();
+  double detectPosAttribute = dyn_cast<ONNXIsInfOp>(op).getDetectPositive();
+
+  // Three different cases: Infinity, Negative Infinity and Positive Infinity
+  bool detectInf = detectPosAttribute == 1 && detectNegAttribute == 1;
+  bool detectNeg = detectNegAttribute == 1 && detectPosAttribute == 0;
+  bool detectPos = detectPosAttribute == 1 && detectNegAttribute == 0;
+
+  Value equPos = create.math.eq(operand, posInf);
+  Value equNeg = create.math.eq(operand, negInf);
+
+  if (detectInf) {
```

Comment:
I can do an `if-else if` here too but from my experience in the community I recall sticking with just `if`

Reply:
nit: I understand they are mutually exclusive. Probably the preferred way would be to return the value directly.
```
if (detectInf) 
  return  create.math.ori(
         create.math.eq(operand, posInf), create.math.eq(operand, negInf));
```
and so on, since result is not used in any ways

Reply:
Done!

Reply:
@AlexandreEichenberger I had to change this back to return the value `result` only because `Windows` check fails with the following error: 

`D:\a\1\onnx-mlir\src\Conversion\ONNXToKrnl\Math\Elementwise.cpp(361) : warning C4715: ‘onnx_mlir::emitScalarOpFor<mlir::ONNXIsInfOp>’: not all control paths return a value`

Since we are directly returning the values in the if's... it is expecting a finally return value.